sycl/doc/extensions/proposed/sycl_ext_oneapi_cuda_cluster_group.asciidoc

```diff
@@ -0,0 +1,404 @@
+have the ability to access one another's local memory, and can be synchronized. 
+This has various applications, convolutions, GEMMs and FFTs to name a few.
+
+This proposal introduces a SYCL API to expose this capability, to be called as
+`cluster_group`, and adds the methods to launch a kernel with a cluster range
+and accessing the various id's associated with the cluster
+launch in device code.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_CUDA_CLUSTER_GROUP` to one of the values defined in the
+table below.  Applications can test for the existence of this macro to determine
+if the implementation supports this feature, or applications can test the
+macro's value to determine which of the extension's features the implementation
+supports
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not version-ed, so the
+ feature-test macro always has this value.
+|===
+
+
+=== Extension to `enum class aspect`
+
+[source]
+----
+namespace sycl {
+enum class aspect {
+  ...
+  sycl_ext_oneapi_cuda_cluster_group
+}
+}
+----
+
+A device requires the `sycl_ext_oneapi_cuda_cluster_group` aspect to
+support launching a kernel with the property defined in the following section.
+
+
+=== Launching a kernel with a `cluster_group`
+
+Because of the special scheduling guarantees associated with a cluster launch,
+the backend must know which kernel would be using this feature. Thus, this 
+proposal introduces a new launch property called as `cluster_size` that will
+contain the cluster in terms of number of work-groups.
+ 
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental::cuda {
+/**
+* Dim Dimensionality of the launch
+* size sycl::range specifying the number of workgroups in the cluster
+  in each dimension.
+*/
+template <int Dim>
+struct cluster_size {
+  cluster_size(const sycl::range<Dim>& size) : 
+    size(size) {}
+  sycl::range<Dim> get_cluster_size();
+  sycl::range<Dim> size;
+};
+using cluster_size_key = cluster_size;
+} // namespace sycl::ext::oneapi::experimental::cuda
+----
+
+The property list can the be constructed as follows - 
+
+[source,c++]
+----
+properties  cluster_launch_property{cluster_size({1, 2, 1})};
+----
+
+Note that the total number of work-groups in the kernel must be a multiple of
+the cluster range in each dimension.
+
+The launch functions introduced in `sycl_ext_oneapi_enqueue_functions` can then
+be used to launch the kernel with the mentioned property.
+
+
+=== Querying Maximum Cluster Size
+To query the maximum supported cluster size, this proposal adds a new device
+query, `max_cluster_group_size`, which returns the maximum possible number of 
+work-groups present inside the cluster.
+
+[source, c++]
+----
+size_t max_cluster_size = 
```

Comment:
We should add a definition for the info descriptor.

Reply:
Added in [beabf](https://github.com/intel/llvm/pull/13594/commits/beabfadeeb7161fb5c5cbc8c69e9cfb9ad5bc888)

Though @AerialMantis , should this be in the info::device namespace as well ?